File: frontend/src/types/adminCatalog.ts
import type { Student } from "@/types/student";

export type AdminSubject = {
  id: number;
  nome: string;
};

export type AdminClass = {
  id: number;
  nome: string;
  identificador?: string | null;
};

export type AdminTeacher = {
  id: number;
  user_id: number;
  nome: string;
  cpf: string;
  email?: string | null;
  ativo: boolean;
};

export type AdminAssignment = {
  id: number;
  teacher_id: number;
  teacher_name: string;
  subject_id: number;
  subject_name: string;
  class_id: number;
  class_name: string;
  class_identifier?: string | null;
};

export type AdminCatalogResponse = {
  subjects: AdminSubject[];
  classes: AdminClass[];
  teachers: AdminTeacher[];
  assignments: AdminAssignment[];
};

export type AdminStudentListResponse = {
  items: Student[];
};

File: backend/app/schemas/admin_catalog.py
from pydantic import BaseModel, Field


class SubjectRead(BaseModel):
    id: int
    nome: str


class SubjectCreate(BaseModel):
    nome: str = Field(min_length=1, max_length=255)


class SubjectUpdate(BaseModel):
    nome: str | None = Field(default=None, max_length=255)


class ClassRead(BaseModel):
    id: int
    nome: str
    identificador: str | None = None


class ClassCreate(BaseModel):
    nome: str = Field(min_length=1, max_length=255)
    identificador: str | None = Field(default=None, max_length=255)


class ClassUpdate(BaseModel):
    nome: str | None = Field(default=None, max_length=255)
    identificador: str | None = Field(default=None, max_length=255)


class TeacherRead(BaseModel):
    id: int
    user_id: int
    nome: str
    cpf: str
    email: str | None = None
    ativo: bool


class AssignmentRead(BaseModel):
    id: int
    teacher_id: int
    teacher_name: str
    subject_id: int
    subject_name: str
    class_id: int
    class_name: str
    class_identifier: str | None = None


class AssignmentCreate(BaseModel):
    teacher_id: int
    subject_id: int
    class_id: int


class CatalogResponse(BaseModel):
    subjects: list[SubjectRead]
    classes: list[ClassRead]
    teachers: list[TeacherRead]
    assignments: list[AssignmentRead]
